.github/workflows/ci.yml

```diff
@@ -84,6 +84,12 @@ jobs:
         with:
           action-merge: fail
           action-fixup: none
+      - name: Check if all commits compile
+        run: |
+          git fetch origin
+          git config user.name "$(git --no-pager log --format=format:'%an' -n 1)"
+          git config user.email "$(git --no-pager log --format=format:'%ae' -n 1)"
+          git rebase origin/master -x '$MAVEN compile -DskipTests'
```

Comment:
Rebasing on origin/master instead of PR head means that the commit hashes for the rebased result may change and users will have harder time to figure out which commit caused compilation failure.

Reply:
There are examples elsewhere in this workflow how to get current PRs head